packages/server/graphql/mutations/dragEstimatingTask.ts

```diff
@@ -48,14 +51,36 @@ export default {
       return standardError(new Error('Meeting phase not found'), {userId: viewerId})
     }
     const {stages} = estimatePhase
-    const draggedStage = stages.find((stage) => stage.id === stageId)
-    if (!draggedStage) {
-      return standardError(new Error('Meeting stage not found'), {userId: viewerId})
+    const taskIds = stages.map(({taskId}) => taskId)
+    const numberOfTasks = new Set(taskIds).size
+    if (newPositionIndex < 0 || newPositionIndex >= numberOfTasks) {
+      return standardError(new Error('Invalid position index'), {userId: viewerId})
     }
+    const draggedStages = stages.filter((stage) => stage.taskId === taskId)
+    if (!draggedStages.length) {
+      return standardError(new Error('No meeting stages were found'), {userId: viewerId})
+    }
+    const stageIds = draggedStages.map((stage) => stage.id)
 
     // RESOLUTION
     // MUTATIVE
-    draggedStage.sortOrder = sortOrder
+    const numberOfDimensions = Math.floor(stages.length / numberOfTasks)
+    const oldPositionIndex = taskIds.indexOf(taskId) / numberOfDimensions
+    let sortOrder
+    if (newPositionIndex === 0) {
+      sortOrder = Math.floor(stages[0].sortOrder / 2)
+    } else if (newPositionIndex === numberOfTasks - 1) {
+      sortOrder = stages[stages.length - 1].sortOrder + ESTIMATE_TASK_SORT_ORDER
+    } else {
+      const offSet = oldPositionIndex > newPositionIndex ? 0 : 1
+      const prevStage = stages[(newPositionIndex + offSet) * numberOfDimensions - 1]
+      const nextStage = stages[(newPositionIndex + offSet) * numberOfDimensions]
+      sortOrder = Math.floor((prevStage.sortOrder + nextStage.sortOrder) / 2)
```

Comment:
-1 this is the only bummer with the client giving us an index vs. it giving us an integer sort order-- we need to find the previous & next stages! Here it's easy because stages live on the Meeting object. If this were an agenda item (or comment, task, reflection, template dimension, template scale, etc.) where each item is its own row in the DB, we'd have to fetch ALL the items to determine where to put it. for those cases, it'd be more efficient to send a single query to the DB that updates every item where the index is greater than this one.

should we leave this as is & worry about setting a correct pattern later or create the new pattern now? up to you!

Reply:
I think we only need to worry about using this integer approach because we need to deal with dimensions, i.e., when we need to move N items and put them in between of two elements and also preserve the ordering. For single dimension items like agenda items, when you do `-SORT_STEP`, `+SORT_STEP` or `(a+b)/2` it's guaranteed the newly generated order is in right position. So I feel it'd be a while before we have another use case that needs this approach so I choose to not optimize earlier.

Reply:
i'm not sure i follow.
could you explain how you could change the sort order of an agenda item without fetching all the agenda items in the meeting?